### ISSUE TITLE
[FEAT]Merge correct_angles script into preprocessing

### ISSUE THREAD

**Issue by DSimonne** [carnisj/bcdi#209]
We could merge the bcdi_correct_angles.py and bcdi_preprocessing.py scripts, many parameters are shared, for the others, one could imagine a boolean parameter "correct" like the "get_temperature" parameter.

**Comment by DSimonne** [comment]
Hi @carnisj, I saw that you started to do some work on that.

I had added some code in the correct_angles script to save the images that are plotted in the script, the rocking curve, and the detector image with a point indicating the Bragg peak:
![central_slice](https://user-images.githubusercontent.com/51970962/145020547-cc9a53cf-3c63-4bc8-a0a2-43797477c48f.png)
![rocking_curve](https://user-images.githubusercontent.com/51970962/145020556-a6fcf8ce-4560-4537-a351-834a9a0f3596.png)
It could be nice to keep that, there were saved in `scan_folder/postprocessing`.

```python
    #################################
    # plot image at Bragg condition #
    #################################
    plt.close()
    plt.imshow(np.log10(abs(data[int(round(z0)), :, :])), vmin=0, vmax=5)
    plt.title(f'Central slice at frame {int(np.rint(z0))}')
    plt.colorbar()

    plt.scatter(bragg_x, bragg_y, color='r', alpha=0.7, linewidth=1)
    plt.savefig(save_dir + "central_slice.png")
    plt.show()
```

```python
    # added plot
    plt.close()
    fig, (ax0, ax1) = plt.subplots(2, 1, sharex='col', figsize=(10, 5))
    ax0.plot(tilt_values, rocking_curve, '.')
    ax0.plot(interp_tilt, interp_curve)
    ax0.axvline(tilt_values[z0], color='r', alpha=0.7, linewidth=1)
    ax0.set_ylabel('Integrated intensity')
    ax0.legend(('data', 'interpolation'))
    ax0.set_title(plot_title)
    ax1.plot(tilt_values, np.log10(rocking_curve), '.')
    ax1.plot(interp_tilt, np.log10(interp_curve))
    ax1.axvline(tilt_values[z0], color='r', alpha=0.7, linewidth=1)

    ax1.set_xlabel('Rocking angle (deg)')
    ax1.set_ylabel('Log(integrated intensity)')
    ax0.legend(('data', 'interpolation'))
    plt.savefig(save_dir + "rocking_curve.png")
    plt.show()
```

I had also renamed pynxraw to preprocessing, it is more constant with how we work.

I had also included a part in which I extracted some data from the script, and then reused it in the workflow.

```python
  # added script
  COM_rocking_curve = tilt_values[z0],
  detector_data_COM = abs(data[int(round(z0)), :, :]),
  try:
      metadata = {
          "tilt_values": tilt_values,
          "rocking_curve": rocking_curve,
          "interp_tilt": interp_tilt,
          "interp_curve": interp_curve,
          "COM_rocking_curve": tilt_values[z0],
          "detector_data_COM": abs(data[int(round(z0)), :, :]),
          "interp_fwhm": interp_fwhm,
          "temperature": temperature,
          "bragg_x": bragg_x,
          "bragg_y": bragg_y,
          "q": q,
          "qnorm": qnorm,
          "dist_plane": dist_plane,
          "bragg_inplane": bragg_inplane,
          "bragg_outofplane": bragg_outofplane,
      }
  except:
      try:
          metadata = {
              "tilt_values": tilt_values,
              "rocking_curve": rocking_curve,
              "interp_tilt": interp_tilt,
              "interp_curve": interp_curve,
              "COM_rocking_curve": tilt_values[z0],
              "detector_data_COM": abs(data[int(round(z0)), :, :]),
              "interp_fwhm": interp_fwhm,
              "bragg_x": bragg_x,
              "bragg_y": bragg_y,
              "q": q,
              "qnorm": qnorm,
              "dist_plane": dist_plane,
              "bragg_inplane": bragg_inplane,
              "bragg_outofplane": bragg_outofplane,
          }
      except Exception as e:
          raise e
  return metadata
```
The only issue I see with the scripts currently, is that they do not return anything. I'm not sure how I could extract these then :/

**Comment by carnisj** [comment]
Hi David, I will try to follow your requests.

For the return value, what we should do is to put which is under "if __name__ == __main__:" in a function. Then you could run the function and not the script anymore (you don't care about the command line arguments, do you?), the difference being that the function can return something. Other people could still run the script and take advantage of the cli.

**Comment by DSimonne** [comment]
Hi Jérome,

OK that's a good idea =D

By the way I'm not sure why i had build two try blocks with the same code inside, I think that for SixS I had included ReadNXS so that I would also get informations like the slits opening, etc ...

I wrote a tiny bit of code at Petra too that would get metadata from the `.fio` files, it is nice to use during the experiment (I put a file in the discord chat if you want to check). The goal was mostly to try to get metadata from the data files and to then create a csv file that has scan metadata (e.g. slits opening, ...) and extracted information like the scattering vector.

![image](https://user-images.githubusercontent.com/51970962/145172164-df6ea508-a5d1-429c-930a-45d0a52a8712.png)

It's quite difficult to implement though since the original data are not nexus files with clear parameters, but I think people are working towards that.

Thanks for the help !
